File: prime-number/haskell/cmd.hs
import System.Environment (getArgs)

firstOrDefault :: [String] -> Int -> Int
firstOrDefault args defaultValue =
    if length args == 1
        then read (head args)
        else defaultValue

getJ :: Int -> Int -> Int -> Int
getJ j i number =
    if i > number
        then j
        else if number `rem` i == 0
                then getJ (j + 1) (i + 1) number
                else getJ j (i + 1) number

getNumber :: Int -> Int -> Int
getNumber primeNumberCount number =
    if primeNumberCount == 0
        then number
        else do
            let numberInc = number + 1
            let j = getJ 0 1 numberInc
            if j == 2
                then getNumber (primeNumberCount - 1) numberInc
                else getNumber primeNumberCount numberInc

main :: IO ()
main = do
    args <- getArgs
    let primeNumberCount = firstOrDefault args 100

    let number = getNumber primeNumberCount 0
    putStrLn ("The latest prime number: " ++ (show number))

File: prime-number/scala/cmd.scala
object cmd extends App
{
    var primeNumberCount = if (args.size == 1) args(0).toInt else 100

    var number = 0
    while (primeNumberCount > 0) {
        number += 1
        var j = 0

        var i = 0
        for (i <- 1 to number) {
            if (number % i == 0) {
                j += 1
            }
        }

        if (j == 2) {
            primeNumberCount -= 1
        }
    }

    println("The latest prime number: " + number)
}
